### PR TITLE
[Refactor] Refactor CmakeLists

### PR DESCRIPTION
**Pull request by dujijun007**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -82,7 +89,7 @@ OPTION(ENABLE_PACKAGING
   OFF)
 OPTION(BUILD_STATIC_LIBRARIES
   "create targets for building static libraries"
-  OFF)
+  ON)
```
Comment (kevincai) [inline CMakeLists.txt]: use -DBUILD_STATIC_LIBRARIES to turn on it, no need to change the default value.

File: CMakeLists.txt [hunk 2]
```diff
@@ -107,6 +114,30 @@ OPTION(ENABLE_COMPILE_TESTS
...
+
+# Set compile flags based on the build type.
+IF ("${CMAKE_BUILD_TYPE}" STREQUAL "ASAN")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CXX_FLAGS_ASAN}")
+ELSEIF ("${CMAKE_BUILD_TYPE}" STREQUAL "LSAN")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CXX_FLAGS_LSAN}")
+ENDIF()
+
+IF (USE_AVX2)
+	SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_AVX2")
```
Comment (kevincai) [inline CMakeLists.txt]: fix the indentiton, keep consistent with the file.

Comment (kevincai) [reply CMakeLists.txt]: where will this `-DUSE_AVX2` option be used? it should be translated to something like `-mavx2`

File: src/shared/CMakeLists.txt [hunk 3]
```diff
@@ -323,25 +325,25 @@ GET_SHARED_FILES(clucene_shared_Files ".")
...
-)
-TARGET_LINK_LIBRARIES(clucene-shared ${CMAKE_THREAD_LIBS_INIT})
-IF ( EXTRA_LIBS )
-  TARGET_LINK_LIBRARIES(clucene-shared ${EXTRA_LIBS})
-ENDIF ( EXTRA_LIBS )
-
-install(TARGETS clucene-shared 
-        DESTINATION ${LIB_DESTINATION} 
-        COMPONENT development )
+IF ( BUILD_SHARED_LIBRARIES )
```
Comment (kevincai) [inline src/shared/CMakeLists.txt:328]: we can tolerant this build shared libraries by default, and do the additional purge in the install path.

Comment (dujijun007) [reply src/shared/CMakeLists.txt:328]: > we can tolerant this build shared libraries by default, and do the additional purge in the install path.

But is it necessary to reserve shared libraries by default? in fact, we will only use static libraries, and adding this config condition will not break the existing build logic.

File: CMakeLists.txt (hunk 2 above)
Comment (dujijun007) [reply CMakeLists.txt]: > where will this `-DUSE_AVX2` option be used? it should be translated to something like `-mavx2`

It's for future use, I will introduce it in next PR.

File: src/shared/CMakeLists.txt (hunk 3 above)
Comment (kevincai) [reply src/shared/CMakeLists.txt:328]: it doesn't break anything, but if the goal can be achieved by not changing the code, why not change it anyway?

**Review by kevincai — APPROVED** [review]

**Pull request merged**